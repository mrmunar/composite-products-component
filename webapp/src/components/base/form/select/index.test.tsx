import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import Select from './';

afterEach(cleanup);

const testData = [
    { id: 1, name: 'item1' },
    { id: 2, name: 'item2' },
    { id: 3, name: 'item3' },
];

test('should take a snapshot', () => {
    const { baseElement } = render(<Select data={testData} />);
    expect(baseElement).toMatchSnapshot();
});

test('initial value is blank', () => {
    const { getByTestId } = render(<Select data={testData} />);
    expect(getByTestId('select')).toHaveValue('');
});

test('can change a value', () => {
    const { getByTestId } = render(<Select data={testData} onChange={() => { }} value="Test Value" />);
    fireEvent.change(screen.getByTestId('select'), {
        target: { value: 2 },
      });
    expect(getByTestId('select')).toHaveValue('2');
});

test('cannot select an item thats not on the list', () => {
    const { getByTestId } = render(<Select data={testData} onChange={() => { }} value="Test Value" />);
    fireEvent.change(screen.getByTestId('select'), {
        target: { value: 10 },
      });
    expect(getByTestId('select')).not.toHaveValue(10);
});

