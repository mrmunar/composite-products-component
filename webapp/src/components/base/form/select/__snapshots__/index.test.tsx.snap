// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <select
      class="custom-select"
      data-testid="select"
    >
      <option
        value=""
      />
      <option
        value="1"
      >
        item1
      </option>
      <option
        value="2"
      >
        item2
      </option>
      <option
        value="3"
      >
        item3
      </option>
    </select>
  </div>
</body>
`;
